src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php

```diff
@@ -58,6 +58,10 @@ public function doHydrate(AbstractMappingEvent $event)
     {
         $document = $event->getDocument();
         if ($document instanceof RedirectTypeBehavior && $document->getRedirectType() !== RedirectType::NONE) {
+            if ($document->getRedirectType() === RedirectType::INTERNAL) {
+                $document->setResourceSegment($document->getRedirectTarget()->getResourceSegment());
```

Comment:
Are you sure this is required? Sorting works without these lines as well (I guess the error was in the persist method), but the page cannot be edited with this line, because you get a duplicated URL.